sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/Analyzer.scala

```diff
@@ -379,6 +381,33 @@ class Analyzer(catalog: Catalog,
     }
   }
 
+  object ResolveSortAlias extends Rule[LogicalPlan] {
+    def apply(plan: LogicalPlan): LogicalPlan = plan transform {
+      case s @ Sort(ordering, global, a @ Aggregate(grouping, aggs, child))
+          if !s.resolved && a.resolved =>
+        val missingInAggs = new ArrayBuffer[NamedExpression]()
+        val aliasedOrdering = ordering.zipWithIndex.map{
+          case (o, i) => {
+            o.child match {
+              case aggOrSub @ (_: AggregateExpression | _: Substring) => {
+                val aliasName = aggOrSub.nodeName + i
+                missingInAggs += Alias(aggOrSub, aliasName)()
+                SortOrder(UnresolvedAttribute(aliasName), o.direction)
```

Comment:
If you are using the `Attribute` then it doesn't matter if the name of the alias conflicts.  The name is only for debugging help after resolution is complete.